atomate/vasp/workflows/base/neb.py

```diff
@@ -0,0 +1,355 @@
+
+from pymatgen.core import Structure
+from pymatgen_diffusion.neb.io import get_endpoints_from_index
+from fireworks.core.firework import Workflow
+
+from atomate.vasp.fireworks.core import NEBFW, NEBRelaxationFW
+from atomate.vasp.firetasks.glue_tasks import TransferNEBTask
+from atomate.utils.utils import get_logger
+
+__author__ = "Hanmei Tang, Iek-Heng Chu"
+__email__ = 'hat003@eng.ucsd.edu, ihchu@eng.ucsd.edu'
+
+logger = get_logger(__name__)
+spec_orig = {"wf_name": "CINEB",
+             "vasp_cmd": ">>vasp_cmd<<",
+             "gamma_vasp_cmd": ">>gamma_vasp_cmd<<",
+             "mpi_command": {"command": "mpirun", "np_tag": "-np"},
+             "ppn": "24",
+             "db_file": ">>db_file<<",
+             "_category": "",
+             "_queueadapter": {"nnodes": 1},
+             "source_dir": os.getcwd(),
+             "dest_dir": os.getcwd(),
+             "structure": None,
+             "path_sites": [],
+             "ini": {},
+             "eps": [{}, {}],
+             "neb": [[{}]]
+             }
+
+
+def _update_spec_from_inputs(spec=None, wf_name=None, structure=None, ini=None, path_sites=None,
+                             endpoints=None, images=None):
+    """
+    Update spec according to inputs. spec pass complete structure information.
+
+    Args:
+        spec(dict): original spec.
+        wf_name (str): a descriptive name.
+        structure (dict): perfect cell structure.
+        path_sites ([int, int]): Indicating pathway site indexes.
+        ini (dict): perfect cell supercell structure.
+        endpoints ([dict]): list of endpoint structures.
+        images ([s0_dict, s1_dict, ...]): list of image_dict, including the two endpoints.
+    """
+    spec = spec or {}
+    s = spec_orig.copy()
+    s.update(spec)
+
+    if structure is not None:
+        assert isinstance(structure, dict)
+        s["structure"] = structure
+
+    if ini is not None:
+        assert isinstance(ini, dict)
+        s["ini"] = ini
+
+    if path_sites is not None:
+        assert isinstance(path_sites, list) and len(path_sites) == 2, "path_sites should be a list!"
+        s["path_sites"] = path_sites
+        st = structure or ini
+        if st is not None:
+            ep0, ep1 = get_endpoints_from_index(Structure.from_dict(st), path_sites)
+            endpoints = [ep0.as_dict(), ep1.as_dict()]
+            s["eps"] = endpoints
+
+    if endpoints is not None:
+        assert len(endpoints) == 2
+        for e in endpoints:
+            assert isinstance(e, dict)
+        s["eps"] = endpoints
+
+    if images is not None:
+        assert isinstance(images, list) and len(images) >= 3
+        for i in images:
+            assert isinstance(i, dict)
+        s["neb"] = [images]
+        n_images = len(images) - 2
+        s["_queueadapter"].update({"nnodes": str(n_images)})
+
+    if wf_name is not None:
+        s["wf_name"] = wf_name
+
+    return s
+
+
+# TODO: Enable setting using >>my_vasp_cmd<<
```

Comment:
I think these are duplicates? Think carefully about whether you need a new method or you can modify an existing one.